### ISSUE TITLE
(t) replication spawn error

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2766]
Thanks to forum member niko for highlighting this issue. Testing branch exhibits the following when attempting to activate a replication receiver:

> Failed to start Replication due to an error: Error running a command. cmd = /opt/rockstor/.venv/bin/supervisorctl start replication. rc = 7. stdout = ['replication: ERROR (spawn error)', '']. stderr = [''] 

```
            Traceback (most recent call last):
  File "/opt/rockstor/src/rockstor/smart_manager/views/replication_service.py", line 82, in post
    superctl(service.name, command)
  File "/opt/rockstor/src/rockstor/system/services.py", line 147, in superctl
    out, err, rc = run_command([SUPERCTL_BIN, switch, service], throw=throw)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/rockstor/src/rockstor/system/osi.py", line 263, in run_command
    raise CommandException(cmd, out, err, rc)
system.exceptions.CommandException: Error running a command. cmd = /opt/rockstor/.venv/bin/supervisorctl start replication. rc = 7. stdout = ['replication: ERROR (spawn error)', '']. stderr = ['']
```

Forum reference: https://forum.rockstor.com/t/failed-to-start-replication-due-to-an-error-error-running-a-command-cmd-opt-rockstor-venv-bin-supervisorctl-start-replication-rc-7-stdout-replication-error-spawn-error-stderr/9136

**Comment by phillxnet** [comment]
supervisord.log for above rpm instance built against an ongoing PR: 5.0.5-2764 running on a Leap 15.5:

```
tail -f /opt/rockstor/var/log/supervisord.log 
...
2023-12-14 17:09:48,073 INFO spawned: 'replication' with pid 6202
2023-12-14 17:09:50,468 INFO success: replication entered RUNNING state, process has stayed up for > than 2 seconds (startsecs)
2023-12-14 17:09:50,468 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:09:50,470 INFO spawned: 'replication' with pid 6288
2023-12-14 17:09:51,758 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:09:52,760 INFO spawned: 'replication' with pid 6373
2023-12-14 17:09:54,022 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:09:56,026 INFO spawned: 'replication' with pid 6458
2023-12-14 17:09:57,302 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:00,308 INFO spawned: 'replication' with pid 6547
2023-12-14 17:10:02,619 INFO success: replication entered RUNNING state, process has stayed up for > than 2 seconds (startsecs)
2023-12-14 17:10:02,620 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:03,623 INFO spawned: 'replication' with pid 6879
2023-12-14 17:10:04,957 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:05,960 INFO spawned: 'replication' with pid 6974
2023-12-14 17:10:07,282 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:09,287 INFO spawned: 'replication' with pid 7078
2023-12-14 17:10:10,821 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:13,827 INFO spawned: 'replication' with pid 7188
2023-12-14 17:10:15,233 INFO exited: replication (exit status 1; not expected)
2023-12-14 17:10:16,235 INFO gave up: replication entered FATAL state, too many start retries too quickly
```
and from the more specific log we have:
```
tail -f /opt/rockstor/var/log/supervisord_replication_stderr.log 
... repeating instances of:

Traceback (most recent call last):
  File "/opt/rockstor/.venv/bin/replicad", line 3, in <module>
    from smart_manager.replication.listener_broker import main
  File "/opt/rockstor/src/rockstor/smart_manager/replication/listener_broker.py", line 27, in <module>
    from sender import Sender
ModuleNotFoundError: No module named 'sender'
```
So we may just have a dangling Python3 style import issue or the like here.

**Comment by phillxnet** [comment]
## Diagnostic notes:

### Receiver
```
rleap15-5:~ # ss -tulpn | grep 10002
tcp   LISTEN 0      100                       192.168.2.199:10002      0.0.0.0:*    users:(("replicad",pid=2505,fd=15))
```

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2777
